src/book/02-projects/01-nubots/03-build-system.mdx

```diff
@@ -353,6 +353,81 @@ To reiterate how the script system loads script files
 
 ## Docker Images
 
+The NUbots codebase uses Docker images to containerise the build process.
+Two images are created, one intended for running code on your local machine named `generic`, and another internded for running code on the robots named `nuc7i7bnh`.
+Apart from flags provided to the build tools and the compiler both images are identical (that is, they contain the same programs and libraries).
+
+---
+
+**NOTE**
+`nuc7i7bnh` is the name Intel has given to the Intel NUC device that the NUgus robots are currently using.
+
+---
+
+The Docker images are cached on [Docker Hub](https://hub.docker.com/r/nubots/nubots) to limit the amount of building that everyone has to do. The cache will be updated whenever the master branch is updated.
+
+The Docker images use [Arch Linux](https://www.archlinux.org/) as their operating system as it is lightweight, flexible, and up-to-date.
+
+The [docker folder](https://github.com/NUbots/NUbots/tree/master/docker) contains the Dockerfile as well as some utility scripts to ease the burden of compiling multiple different libraries. Other files in this folder end up inside of the image and control either the behaviour of the compiler or build system, or the runtime behaviour of the libraries that are built.
+
+If you need to add a new library of program to the Docker image you should add it to the end of the Dockerfile to minimise the amount of rebuilding that needs to happen. Look for the following marker
+
+```sh
+#######################################
+### ADD NEW PROGRAMS/LIBRARIES HERE ###
+#######################################
+```
+
+### Adding a new library to the robot toolchain
+
+If the intention is to have the new library running on the robot then you need to build the library from source. If the library you are building is well-behaved then it should be a simple matter of add a line like the following to the Dockerfile
+
+```docker
+RUN install-from-source http://url.to.the.source.tarball.com
+```
+
+---
+
+**SPOILER**
+Libraries are rarely ever well-behaved
```

Comment:
Lol, but to be fair the majority of our libraries are well behaved (by majority I mean >50%). We don't want to scare people off too much from adding new libraries if they need them